google/cloud/forseti/services/model/importer/importer.py

```diff
@@ -341,8 +343,15 @@ def group_name(parent):
                 type=m_type,
                 member_name=name)
 
-        self._membership_cache.append(
-            (group_name(parent), member))
+        parent_group = group_name(parent)
+
+        if parent_group not in self.membership_cache:
+            self.membership_cache[parent_group] = set()
```

Comment:
what is the purpose of membership_cache? Also to avoid duplication?

Reply:
Yes, we can have duplicate calls due to members with different casing. 
e.g. (group/QWE, user/ABC) and (group/qwe, user/abc) will both get appended to the membership list if we don't check for duplicate.

Reply:
In this case, I would say rename it. For example, membership_cache -> membership_graph, membership -> membership_items. The name can be more descriptive.